plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +358,27 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*10.
+		if checkOverFlow(totalPriority, config.Weight*10) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * 10
+	}
 	return configs, nil
 }
 
+// checkOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkOverFlow(totalPriority int, weight int) bool {
```

Comment:
I guess, if we want to use this in someother location, its better to generalize the name to `checkIntegerOverflow` as this function can check for any 2 int's overflow.